simulationdataschema/model_system.py

```diff
@@ -528,10 +571,134 @@ class ModelSystem(System):
         """,
     )
 
-    atomic_cell = SubSection(sub_section=AtomicCell.m_def, repeats=True)
+    def resolve_chemical_formulas(self, formula):
+        """
+        Resolves the chemical formulas of the ModelSystem in different formats.
 
-    symmetry = SubSection(sub_section=Symmetry.m_def, repeats=True)
+        Args:
+            formula (Formula): The Formula object from NOMAD atomutils containing the
+            chemical formulas.
+        """
+        self.descriptive = formula.format("descriptive")
+        self.reduced = formula.format("reduced")
+        self.iupac = formula.format("iupac")
+        self.hill = formula.format("hill")
+        self.anonymous = formula.format("anonymous")
+
+    def normalize(self, archive, logger):
+        atomic_cell = check_parent_and_atomic_cell(self, logger)
+        ase_atoms = atomic_cell.to_ase_atoms(logger)
+        formula = None
+        try:
+            formula = Formula(ase_atoms.get_chemical_formula())
+            # self.chemical_composition = ase_atoms.get_chemical_formula(mode="all")
+        except ValueError as e:
+            logger.warning(
+                "Could not extract the chemical formulas information.",
+                exc_info=e,
+                error=str(e),
+            )
+        if formula:
+            self.resolve_chemical_formulas(formula)
+            self.m_cache["elemental_composition"] = formula.elemental_composition()
+
+
+class ModelSystem(System):
+    """
+    Model system used as an input for the computation. It inherits from `System` where a set
+    of sub-sections for the `elemental_composition` is defined.
+
+    We also defined:
+        - `name` refers to all the verbose and user-dependent naming in ModelSystem,
+        - `type` refers to the type of the ModelSystem (atom, bulk, surface, etc.),
+        - `dimensionality` refers to the dimensionality of the ModelSystem (0D, 1D, 2D, 3D),
+
+    If the ModelSystem `is_representative`, the normalization occurs. The time evolution of
```

Comment:
So did we decide that only 1 system can be representative? Do we need a check for this if so?

Reply:
Let's branch this out to its own discussion: https://github.com/FAIRmat-NFDI/AreaC-Meetings/discussions/34